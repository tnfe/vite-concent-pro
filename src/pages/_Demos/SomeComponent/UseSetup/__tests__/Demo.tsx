import React from 'react';
import { mount } from 'enzyme';
import Demo from '../index';

// 组件实例
let ins = null as any;

/**
 * @author fancyzhong
 * @priority P0
 * @casetype unit
 */
describe('Demo Render', () => {
  beforeAll(() => {
    const mockRouterProps = { history: {} as any, location: {} as any, match: {} as any };
    ins = mount(<Demo {...mockRouterProps} />);
  });

  test('generate Demo snapshot', () => {
    expect(ins).toMatchSnapshot();
  });
});
