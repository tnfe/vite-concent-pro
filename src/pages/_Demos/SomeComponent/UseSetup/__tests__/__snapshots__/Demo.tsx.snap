// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo Render generate Demo snapshot 1`] = `
<Memo(DemoPageTodoList)
  history={Object {}}
  location={Object {}}
  match={Object {}}
>
  <DemoUseSetup>
    <div>
      <input
        onChange={[Function]}
        value="xx"
      />
      <h1>
        reverse: 
        xx
      </h1>
      <h1>
        2
      </h1>
      <Button
        onClick={[Function]}
        type="primary"
      >
        <Wave>
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="button"
          >
            <LoadingIcon
              existIcon={false}
              loading={false}
              prefixCls="ant-btn"
            >
              <CSSMotion
                motionName="ant-btn-loading-icon-motion"
                onAppearActive={[Function]}
                onAppearStart={[Function]}
                onEnterActive={[Function]}
                onEnterStart={[Function]}
                onLeaveActive={[Function]}
                onLeaveStart={[Function]}
                removeOnLeave={true}
                visible={false}
              >
                <DomWrapper />
              </CSSMotion>
            </LoadingIcon>
            <span
              key=".0"
            >
              toogleIsOk true
            </span>
          </button>
        </Wave>
      </Button>
    </div>
  </DemoUseSetup>
</Memo(DemoPageTodoList)>
`;
